chromatic_fitting.py

```diff
@@ -214,40 +372,43 @@ def attach_models(self, models):
             if isinstance(model, LightcurveModel):
                 # make a "copy" of each model:
                 new_model = model.__class__()
-                new_model.initialize_empty_model()
+                # new_model.initialize_empty_model()
+                new_model.pymc3_model = self.pymc3_model
                 # can't just do new_v.setup_parameters(**dict(v.parameters)) - runs into weird pymc3 inheritance
                 # issues!
                 model_params = {}
                 # for every parameter in the separate models redefine them in the separate and new models
                 for k, v in model.parameters.items():
-                    if isinstance(v, Fixed):
-                        # parameter is Fixed
-                        model_params[k] = v.__class__(v.value)
-                    elif isinstance(v, WavelikeFixed):
+                    if isinstance(v, WavelikeFixed):
                         # parameter is WavelikeFixed
                         model_params[k] = v.__class__(v.values)
+                    elif isinstance(v, Fixed):
+                        # parameter is Fixed
+                        model_params[k] = v.__class__(v.value)
                     else:
                         # parameter is Fitted or WavelikeFitted
                         model_params[k] = v.__class__(v.distribution, **v.inputs)
                 # set up parameters in new models
                 new_model.setup_parameters(**model_params)
                 new_models[name] = new_model
-                all_params = all_params | model_params # what happens if same keys?
+                # all_params = all_params | model_params # what happens if same keys?
             else:
                 print("This class can only be used to combine LightcurveModels!")
 
         # set up parameters in new combined model
-        self.setup_parameters(**all_params)
-        self.models_combined = new_models
+        # self.setup_parameters(**all_params)
+        self.chromatic_models = new_models
 
     def apply_operation_to_constituent_models(self, operation, *args):
```

Comment:
You don't need it yet, but at some point you might need a `**kwargs` option here too!

Reply:
Added